From a75e29a12d5cf854805ce195578901bfccb17f44 Mon Sep 17 00:00:00 2001
From: Jennifer Herbert <jennifer.herbert@citrix.com>
Date: Tue, 4 Dec 2018 13:47:48 +0000
Subject: [PATCH] Revert "rtc: cmos: Do not assume irq 8 for rtc when there are
 no legacy irqs"

This reverts commit a1e23a42f1bdc00e32fc4869caef12e4e6272f26.

Conflicts:
	drivers/rtc/rtc-cmos.c
---
 drivers/rtc/rtc-cmos.c | 14 ++++----------
 1 file changed, 4 insertions(+), 10 deletions(-)

diff --git a/drivers/rtc/rtc-cmos.c b/drivers/rtc/rtc-cmos.c
index df0c577..f7cf01d 100644
--- a/drivers/rtc/rtc-cmos.c
+++ b/drivers/rtc/rtc-cmos.c
@@ -1293,23 +1293,17 @@ static int cmos_pnp_probe(struct pnp_dev *pnp, const struct pnp_device_id *id)
 {
 	cmos_wake_setup(&pnp->dev);
 
-	if (pnp_port_start(pnp, 0) == 0x70 && !pnp_irq_valid(pnp, 0)) {
-		unsigned int irq = 0;
-#ifdef CONFIG_X86
+	if (pnp_port_start(pnp, 0) == 0x70 && !pnp_irq_valid(pnp, 0))
 		/* Some machines contain a PNP entry for the RTC, but
 		 * don't define the IRQ. It should always be safe to
-		 * hardcode it on systems with a legacy PIC.
+		 * hardcode it in these cases
 		 */
-		if (nr_legacy_irqs())
-			irq = 8;
-#endif
 		return cmos_do_probe(&pnp->dev,
-				pnp_get_resource(pnp, IORESOURCE_IO, 0), irq);
-	} else {
+				pnp_get_resource(pnp, IORESOURCE_IO, 0), 8);
+	else
 		return cmos_do_probe(&pnp->dev,
 				pnp_get_resource(pnp, IORESOURCE_IO, 0),
 				pnp_irq(pnp, 0));
-	}
 }
 
 static void cmos_pnp_remove(struct pnp_dev *pnp)
-- 
1.8.3.1

